cpp/src/parquet/reader_test.cc

```diff
@@ -502,6 +502,25 @@ TEST_F(TestAllTypesPlain, ColumnSelectionOutOfRange) {
   ASSERT_THROW(printer2.DebugPrint(ss, columns), ParquetException);
 }
 
+// Tests getting a record reader from a row group reader.
+TEST(TestFileReader, GetRecordReader) {
+  ReaderProperties reader_props;
+  reader_props.enable_read_dense_for_nullable();
+  std::unique_ptr<ParquetFileReader> file_reader = ParquetFileReader::OpenFile(
+      alltypes_plain(), /* memory_map = */ false, reader_props);
+  std::shared_ptr<RowGroupReader> group = file_reader->RowGroup(0);
+
+  std::shared_ptr<internal::RecordReader> col_record_reader_ = group->RecordReader(0);
+
+  ASSERT_TRUE(col_record_reader_->read_dense_for_nullable());
```

Comment:
Does this really test `read_dense_for_nullable()`? 🤔

Reply:
The goal here is to test if the option is passed down to the underlying record reader. I added a separate more comprehensive test.

Reply:
Okay